recipe/meta.yaml

```diff
@@ -34,55 +31,34 @@ requirements:
     - {{ compiler('rust') }}
     - cmake
     - make
-    - libprotobuf
+    - nodejs
+    - pnpm
+    - pkg-config
+    - git
+    - wget # to download boost sources from install_tools.mjs
+    - coreutils # sed, build.sh
+    # XXX(tom): emsdk? no, use `jupyterlab` tooling to build extension from npm source (which will DL wasm blob)
+    - xsimd # XXX(tom): Andrew says we shouldn't need this b/c we disable SIMD in arrow build
+    - lz4 # XXX(tom): likewise, this should be built from source downloaded by psp repo script. and yet the build fails without it
+    - nvim # XXX(tom): deleteme
   host:
     - python
     - pip
     - maturin >=1.4,<2.0
-    - libboost-devel
+    - libstdcxx
```

Comment:
this is questionable for ABI compatibility reasons

Reply:
arrow/parquet went back and forth static-linking libstdc++ and eventually abandoned it, "static linking libstdc++ is far too finicky". see https://github.com/apache/parquet-cpp/pull/133#issuecomment-231516828 

Just to get a picture of what things are like in Conda on the legacy systems we must support, I created a Rocky Linux 8 docker image with Miniconda3 installed, then created a Conda environment there and installed `arrow-cpp`. This installed `libstdcxx-ng` from Conda, and the .so is dynamically linked against it with the newest symbol version being GLIBCXX_3.4.29

```
(arrow-test) [root@33111abbeccd ~]# objdump -T /miniconda/envs/arrow-test/lib/libarrow.so | grep GLIBCXX | awk '// { print $5 }' | sort -V | uniq
GLIBCXX_3.4
GLIBCXX_3.4.5
GLIBCXX_3.4.9
GLIBCXX_3.4.11
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_3.4.20
GLIBCXX_3.4.21
GLIBCXX_3.4.22
GLIBCXX_3.4.26
GLIBCXX_3.4.29
```

For maximum support, I think we can do this:

1. Identify the earliest libstdc++ version we want to support
2. In our recipe, pin the host dependency on libstdcxx to that version
3. Require versions newer than that as runtime dependencies

I have looked around at similar libraries recipes (arrow-cpp, tensorflow), and they don't seem to do anything like this: on Linux, it looks like those recipes transitively depend on libstdcxx or libstdcxx-ng without any particular version constraints.

Reply:
> I have looked around at similar libraries recipes (arrow-cpp, tensorflow), and they don't seem to do anything like this

this is not a good sign for this PR